plone/app/upgrade/utils.py

```diff
@@ -25,13 +25,13 @@
 
 logger = logging.getLogger("plone.app.upgrade")
 
-plone_version = pkg_resources.get_distribution("Products.CMFPlone").version
+plone_version = distribution("Products.CMFPlone").version
 
 
 def version_match(target):
     """Given, our versioning scheme is always major.minorANYTHING, where major
     and minor are single-digit numbers, we can compare versions as follows.
-    pkg_resources.parse_version is not compatible with our versioning scheme
+    packaging.version.parse is not compatible with our versioning scheme
     (like '5.0b1') and also not compatible with the semver.org proposal
     (requires '5.0-beta1').
     """
```

Comment:
Is the comment still accurate? packaging.version.parse seems to handle parsing `5.0b1` fine.

Reply:
We could change the comment to this:

```
    """Does a target version have the same major and minor version as CMFPlone?

    Given, our versioning scheme is always major.minor.ANYTHING, where major
    and minor are single-digit numbers, we can compare versions as follows,
    without worrying about strictly following semantic versioning.

    TODO This will not work in Plone 10.
    """
```

But `version_match` is not used in current Plone.  So we could also add a deprecation warning that we will remove this function in Plone 7.

Reply:
Thanks, I was going a bit fast on the replacements and did not give too much thought on each case 😅 

@mauritsvanrees I added your comment and the deprecation warning, so I removed the last line, as we will not reach Plone with this function